src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -0,0 +1,843 @@
+            helper.AddOrInsertItemTest(col, "four");
+        }
+
+        /// <summary>
+        /// Tests that it is possible to remove an item from the collection.
+        /// - Removing an item from the collection results in a false.
+        /// - Removing null from collection returns a false.
+        /// - Removing an item that has duplicates only takes out the first instance.
+        /// </summary>
+        [Fact]
+        public static void RemoveTest()
+        {
+            // trying to remove item in collection.
+            string[] anArray = { "one", "two", "three", "four" };
+            ObservableCollection<string> col = new ObservableCollection<string>(anArray);
+            CollectionAndPropertyChangedTester helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, 2, "three", true, hasDuplicates: false);
+
+            // trying to remove item not in collection.
+            anArray = new string[] { "one", "two", "three", "four" };
+            col = new ObservableCollection<string>(anArray);
+            helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, -1, "three2", false, hasDuplicates: false);
+
+            // removing null
+            anArray = new string[] { "one", "two", "three", "four" };
+            col = new ObservableCollection<string>(anArray);
+            helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, -1, null, false, hasDuplicates: false);
+
+            // trying to remove item in collection that has duplicates.
+            anArray = new string[] { "one", "three", "two", "three", "four" };
+            col = new ObservableCollection<string>(anArray);
+            helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, 1, "three", true, hasDuplicates: true);
+            // want to ensure that there is one "three" left in collection and not both were removed.
+            int occurancesThree = 0;
+            foreach (var item in col)
+            {
+                if (item.Equals("three"))
+                    occurancesThree++;
+            }
+            Assert.Equal(1, occurancesThree);
+        }
+
+        /// <summary>
+        /// Tests that a collection can be cleared.
+        /// </summary>
+        [Fact]
+        public static void ClearTest()
+        {
+            string[] anArray = { "one", "two", "three", "four" };
+            ObservableCollection<string> col = new ObservableCollection<string>(anArray);
+
+            col.Clear();
+            Assert.Equal(0, col.Count);
+            foreach (var item in col)
+            {
+                Assert.True(false, "Should not be able to iterate through an empty collection.");
+            }
+
+            try
+            {
+                string g = col[1];
+                Assert.True(false, "Should not have been able to get an item from an empty collection. Guid g: " + g);
```

Comment:
This (and other occurrences of this pattern) can be changed to use Assert.Throws